src/helpers/spaces.ts

```diff
@@ -145,6 +145,57 @@ async function getVotes() {
   return await db.queryAsync(query);
 }
 
+export async function getCombinedMembersAndVoters(
+  spaceId,
+  cursor = null,
+  pageSize
+) {
+  const params = [spaceId, spaceId, spaceId, spaceId];
+  const query = `
+    SELECT address
+    FROM (
+      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.admins[*]')) AS address 
+       FROM spaces 
+       WHERE id = ?)
+      UNION
+      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.moderators[*]')) AS address 
+       FROM spaces 
+       WHERE id = ?)
+      UNION
+      (SELECT DISTINCT JSON_UNQUOTE(JSON_EXTRACT(settings, '$.members[*]')) AS address 
+       FROM spaces 
+       WHERE id = ?)
+      UNION
+      (SELECT DISTINCT voter AS address
+       FROM votes
+       WHERE space = ?)
```

Comment:
should limit to 500 voters

Reply:
I have placed the limit to the Outer Query, using the param, pageSize.
Placing a limit on the inner query would mess up the pagination, wouldn't it?

Reply:
Yes but this is taking a lot of time, especially for spaces with a lot of votes, we don't need to read all the distinct voters here, we just need 500 of them, 

you can remove pagination from the outer query

Also for `admins`, `moderators` and `members` -> all this data is already available on `space` object, we don't need to read them from database again,

Reply:
We made an improvement on this, now the with cursor filter,  if cursor is present, admins, moderators, and members are excluded from the distinct voter list. Also regardless we are not querying admins, moderators and members.

Query now only queries the Votes Table hence eliminating the need for subqueries.

Reply:
how to use cursor now? you have an example request? i tried few ways but couldn't get it working

Reply:
A request should be something like` /ens.eth/members?cursor=983434323` ( Cursor Example )

Could you share what message do you get?
Could we get on a call and debug this? As I don't have a way to see whats wrong, its taking longer to close this issue